client/my-sites/checkout/composite-checkout/hooks/use-stored-cards.ts

```diff
@@ -66,7 +66,11 @@ export default function useStoredCards(
 function storedCardsReducer( state: StoredCardState, action: StoredCardAction ) {
 	switch ( action.type ) {
 		case 'FETCH_END':
-			return { ...state, storedCards: action.payload, isLoading: false };
+			// The endpoint may have returned a non-array without actually failing,
+			// so we need to ensure that storedCards is actually an array here.
+			// eslint-disable-next-line no-case-declarations
+			const storedCards = Array.isArray( action.payload ) ? action.payload : [];
```

Comment:
Minor, but since this creates a variable, it's best to create a block scope around this case statement to prevent the variable leaking into the rest of the function. You can do this by adding curly braces around the body of the `case` after the colon. That will also allow removing this `eslint-disable-next-line` comment.